firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionDetailsViewController.swift

```diff
@@ -0,0 +1,417 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import Shared
+import SiteImageView
+import ComponentLibrary
+
+struct TrackingProtectionDetailsModel {
+    let topLevelDomain: String
+    let title: String
+    let URL: String
+
+    let getLockIcon: (ThemeType) -> UIImage
+    let connectionStatusMessage: String
+    let connectionSecure: Bool
+
+    let viewCertificatesButtonTitle: String = .Menu.EnhancedTrackingProtection.viewCertificatesButtonTitle
+
+    let detailsViewA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.DetailsScreen.mainView
+    let connectionViewA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.DetailsScreen.connectionView
+    let headerViewA11yId = AccessibilityIdentifiers.EnhancedTrackingProtection.DetailsScreen.headerView
+    let containerViewA11Id = AccessibilityIdentifiers.EnhancedTrackingProtection.DetailsScreen.containerView
+    let viewCertButtonA11Id = AccessibilityIdentifiers.EnhancedTrackingProtection.DetailsScreen.certificatesButton
+}
+
+class TrackingProtectionDetailsViewController: UIViewController, Themeable {
+    // MARK: - UI
+    private let scrollView: UIScrollView = .build { scrollView in }
+    private let baseView: UIView = .build { view in }
+    private let siteTitleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.numberOfLines = 2
+        label.accessibilityTraits.insert(.header)
+    }
+
+    // MARK: Header View
+    private let headerView = UIView()
+    private let horizontalLine: UIView = .build { _ in }
+    private var closeButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * TPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.imageView?.contentMode = .scaleAspectFit
+        button.setImage(UIImage(named: StandardImageIdentifiers.Medium.cross), for: .normal)
+    }
+    private var backButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * TPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.setTitle(.KeyboardShortcuts.Back, for: .normal)
+        button.setImage(UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.chevronLeft)
+            .withRenderingMode(.alwaysTemplate),
+                        for: .normal)
+        button.titleLabel?.font = TPMenuUX.Fonts.viewTitleLabels.scaledFont()
+    }
+
+    // MARK: Connection Status View
```

Comment:
For readability, what do you think about separating these components into its own UIView to avoid this VC from getting to large. i.e. `ConnectionStatusView` and `CertificatesView` and `VerifiedByView`. Similar to how `MicrosurveyConfirmationView` is separate from the `MicrosurveyViewController`. This cleans up code in a more readable and organized fashion.

Reply:
added some custom views for the header, status and the verified by sections, the see certificates view is just a single button.

Reply:
okay thank you! I added some comments in terms of adding some more code from the VC to those views so they can be more self contained and decoupled from the VC